testsuite/shared/src/main/java/org/jboss/as/test/integration/management/cli/CliProcessWrapper.java

```diff
@@ -0,0 +1,372 @@
+            } catch (InterruptedException e) {
+                fail("Interrupted");
+            }
+
+            waitingTime += resultInterval;
+
+            try{
+                cliProcess.exitValue();
+                wait = false;
+                closed = true;
+            } catch(IllegalThreadStateException e) {
+                // Windows check...
+                if(cliOutputBuffer.toString().endsWith("Press any key to continue . . . ")){
+                    pushLineAndWaitForClose("l");
+                }
+                // cli still working
+            }
+
+            // If the timeout is reached, destroy the process and return false
+            if (waitingTime > resultTimeout) {
+                cliProcess.destroy();
+                wait = false;
+            }
+        }
+
+        if(logErrors && !closed){
+            logErrors();
+        }
+
+        return closed;
+    }
+
+    private boolean outputHasPrompt(String prompt){
+        if(cliOutputBuffer.toString().contains("\n")) {
+            if (prompt == null) {
+                return cliOutputBuffer.toString().substring(cliOutputBuffer.toString().lastIndexOf("\n")+1).matches(".*[\\[].*[\\]].*");
+            } else {
+                return cliOutputBuffer.toString().substring(cliOutputBuffer.toString().lastIndexOf("\n")).contains(prompt);
+            }
+        }else{
+            if (prompt == null) {
+                return cliOutputBuffer.toString().matches(".*[\\[].*[\\]].*");
+            } else {
+                return cliOutputBuffer.toString().contains(prompt);
+            }
+        }
+    }
+
+    private class CliResultsReader implements Runnable {
+
+        InputStream cliStream = null;
+
+        public CliResultsReader(InputStream inputStream){
+            cliStream = inputStream;
+        }
+
+        @Override
+        public void run() {
+            java.io.InputStreamReader isr = new java.io.InputStreamReader(cliStream);
+            java.io.BufferedReader br = new java.io.BufferedReader(isr);
+
+            try{
+                int intCharacter = 0;
+                Character character;
+                // While the next character is a valid character and isn't null
+                while ((intCharacter = br.read()) != -1 && (character = (char)intCharacter) != null) {
+                    cliOutputBuffer.append(character);
+                }
+            } catch (IOException e) {
+                fail("Failed to read commands output: " + e.getLocalizedMessage());
+            }
+        }
+
+    }
+
+    private class CliErrorsReader implements Runnable {
+
+        InputStream cliStream = null;
+
+        public CliErrorsReader(InputStream inputStream){
+            cliStream = inputStream;
+        }
+
+        @Override
+        public void run() {
+            java.io.InputStreamReader isr = new java.io.InputStreamReader(cliStream);
+            java.io.BufferedReader br = new java.io.BufferedReader(isr);
```

Comment:
this two are never closed,
probably best if you do add it to try-with-resoruces block